.github/workflows/ci.yaml

```diff
@@ -0,0 +1,12 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
Yup - this is subject to change. Whilst we're getting CI configured correctly for the new infra, just minimising pointless GHA runs. It's a fair bit of work to get CI on GHA to hook up to our CD pipelines with our existing infra (Jenkins) - so our existing CI will be running probably for a while yet (so still be getting feedback for pushed commits not raised as a PR).

That aside - I think the user need "of being able to test a specific individual commit without a PR" could be better met by manually triggering CI workflow, rather than just automatically running it for every commit pushed. I'd speculate that vast majority CI runs for pushed commits (that aren't raised as a PR) won't actually be used and just be a waste. But we don't need to make that decision yet.